### PR TITLE
Add optional propagation tests

### PR DESCRIPTION
**Pull request by Yas-Franco**
For issue #89 
The b3 and jaeger propagation formats are not required. Tests have been added in the group "optional-tests". The other teats are also put in to groups and the ReadMe.adoc has been updated with instructions on including these

### CODE REVIEW

File: tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/TestApplication.java [hunk 1]
```diff
@@ -51,7 +53,7 @@ public static WebArchive createDeployment() {
         return ShrinkWrap.create(WebArchive.class);
     }
 
-    @Test
+    @Test(groups = TEST_GROUP_REST)
```
Comment (Emily-Jiang) [inline tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/TestApplication.java:56]: Forcing new tests to specify group is a big ask.

Comment (yasmin-aumeeruddy) [reply tracing/tck/src/main/java/org/eclipse/microprofile/telemetry/tracing/tck/TestApplication.java]: I have changed this now so 'optional-tests' is the only group and the ReadMe is up to date

File: tracing/tck/README.adoc [hunk 2]
```diff
@@ -56,15 +56,43 @@ You also need to specify which tests you want to run, e.g. create a file `tck-su
...
+[source, xml]
+----
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
+
+<suite name="microprofile-telemetry-tracing-TCK" verbose="2" configfailurepolicy="continue" >
+    <test name="telemetry-tracing-tests" verbose="10">
+        <groups>
+            <run>
+                <!-- The required REST and CDI tests that all MP-TELEMETRY implementations must pass -->
+                <include name="cdi"/>
```
Comment (Emily-Jiang) [inline tracing/tck/README.adoc:86]: I don't think you defined these groups? Can you use exclude?

Comment (yasmin-aumeeruddy) [reply tracing/tck/README.adoc]: This was from an older commit. I have squashed them to avoid confusion

**Pull request merged**